### DISCUSSION TITLE
onDispose functionality

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-rs#71] [General]
Seems that `use.effect` Dart counterpart seems difficult to implement in Rust, but I'm thinking only the onDispose part of it. Do you think it could be feasible for me to implement it? If so, it would be necessary a `Container.dispose`.

**Comment by GregoryConrad** [comment]
The effects that have the dependencies list (use.effect, use.memo, etc.) tend to not map as well to Rust. I could implement them, but they might be a little funky (which is why I haven't implemented them yet).

> but I'm thinking only the onDispose part of it

No problem there, use `use.value` with a type that implements `Drop`. I made a type that is used for this very purpose here (feel free to copy paste `FunctionalDrop` into your own code):
https://github.com/GregoryConrad/rearch-rs/blob/710ae301ceea76ff88f43e8d52423ebc19b086fa/rearch-tokio/src/lib.rs#L6

Whenever you change the value inside the `use.value` to a new instance of `FunctionalDrop`, the old one will be executed (since it'll be dropped), thus mimic-ing `onDispose`.

**Reply by busslina** [reply]
Cool, thanks !
